satellite.go

```diff
@@ -0,0 +1,214 @@
+package main
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/briandowns/spinner"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+)
+
+type SatellitePingResponse struct {
+	Results struct {
+		Foreman struct {
+			Database struct {
+				Active     bool   `json:"active"`
+				DurationMs string `json:"duration_ms"`
+			} `json:"database"`
+			Cache struct {
+				Servers []struct {
+					Status     string `json:"status"`
+					DurationMs string `json:"duration_ms"`
+				} `json:"servers"`
+			} `json:"cache"`
+		} `json:"foreman"`
+		Katello struct {
+			Services struct {
+				Candlepin struct {
+					Status     string `json:"status"`
+					DurationMs string `json:"duration_ms"`
+				} `json:"candlepin"`
+				CandlepinAuth struct {
+					Status     string `json:"status"`
+					DurationMs string `json:"duration_ms"`
+				} `json:"candlepin_auth"`
+				ForemanTasks struct {
+					Status     string `json:"status"`
+					DurationMs string `json:"duration_ms"`
+				} `json:"foreman_tasks"`
+				KatelloEvents struct {
+					Status     string `json:"status"`
+					Message    string `json:"message"`
+					DurationMs string `json:"duration_ms"`
+				} `json:"katello_events"`
+				CandlepinEvents struct {
+					Status     string `json:"status"`
+					Message    string `json:"message"`
+					DurationMs string `json:"duration_ms"`
+				} `json:"candlepin_events"`
+				Pulp3 struct {
+					Status     string `json:"status"`
+					DurationMs string `json:"duration_ms"`
+				} `json:"pulp3"`
+				Pulp3Content struct {
+					Status     string `json:"status"`
+					DurationMs string `json:"duration_ms"`
+				} `json:"pulp3_content"`
+			} `json:"services"`
+			Status string `json:"status"`
+		} `json:"katello"`
+	} `json:"results"`
+}
+
+// ConfigureSatelliteResult is structure holding information about results
+// of configuring host from Satellite server. The result could be printed
+// in machine-readable format.
+type ConfigureSatelliteResult struct {
+	Hostname                 string `json:"hostname"`
+	HostnameError            string `json:"hostname_error,omitempty"`
+	IsServerSatellite        bool   `json:"is_server_satellite_running"`
+	SatelliteServerHostname  string `json:"satellite_server_hostname"`
+	SatelliteServerScriptUrl string `json:"satellite_server_script_url"`
+	HostConfigured           bool   `json:"host_configured"`
+	format                   string
+}
+
+// Error implement error interface for structure ConfigureSatelliteResult
+func (configureSatelliteResult ConfigureSatelliteResult) Error() string {
```

Comment:
There is a Go convention to refer to the 'self' object in a method as a single
or small variable name. I think this could be made a bit more concise by
replacing `configureSatelliteResult` with `result`. Maybe then rename the
existing `result` string to `message` or something like that.